windows/win_iis_webapppool.ps1

```diff
@@ -90,10 +90,14 @@ try {
       Stop-WebAppPool -Name $name -ErrorAction Stop
       $result.changed = $TRUE
     }
-    if ((($state -eq 'started') -and ($pool.State -eq 'Stopped')) -or ($state -eq 'restarted')) {
+    if ((($state -eq 'started') -and ($pool.State -eq 'Stopped'))) {
       Start-WebAppPool -Name $name -ErrorAction Stop
       $result.changed = $TRUE
     }
+    if ($state -eq 'restarted') {
+      Restart-WebAppPool -Name $name
```

Comment:
This doesn't work if the pool is stopped. You basically want to start the pool if it's stopped and requested state is started or restarted, otherwise if requested state is restarted, restart it.